src/cueUtilities.js

```diff
@@ -196,139 +224,168 @@ module.exports = function (params, cy, api) {
-		});
-		cy.on('mouseup', 'node', data.eMouseUp = function(e){
-			currMousePos = e.renderedPosition || e.cyRenderedPosition
-		});
-
-		cy.on('grab', 'node', data.eGrab = function (e) {
-			preventDrawing = true;
-		});
-
-		cy.on('free', 'node', data.eFree = function (e) {
-			preventDrawing = false;
-		});
-
-		cy.on('position', 'node', data.ePosition = function () {
-			if (nodeWithRenderedCue)
-				clearDraws();
-		});
-
-		cy.on('remove', 'node', data.eRemove = function () {
-			clearDraws();
-			nodeWithRenderedCue = null;
-		});
-
-		var ur;
-		cy.on('select', 'node', data.eSelect = function(){
-			if (this.length > cy.nodes(":selected").length)
-				this.unselect();
-		});
-
-		cy.on('tap', 'node', data.eTap = function (event) {
-			var node = nodeWithRenderedCue;
-      var opts = options();
-			if (node){
-				var expandcollapseRenderedStartX = node._private.data.expandcollapseRenderedStartX;
-				var expandcollapseRenderedStartY = node._private.data.expandcollapseRenderedStartY;
-				var expandcollapseRenderedRectSize = node._private.data.expandcollapseRenderedCueSize;
-				var expandcollapseRenderedEndX = expandcollapseRenderedStartX + expandcollapseRenderedRectSize;
-				var expandcollapseRenderedEndY = expandcollapseRenderedStartY + expandcollapseRenderedRectSize;
-                
-                var cyRenderedPos = event.renderedPosition || event.cyRenderedPosition;
-				var cyRenderedPosX = cyRenderedPos.x;
-				var cyRenderedPosY = cyRenderedPos.y;
-				var factor = (opts.expandCollapseCueSensitivity - 1) / 2;
-
-				if ( (Math.abs(oldMousePos.x - currMousePos.x) < 5 && Math.abs(oldMousePos.y - currMousePos.y) < 5)
-					&& cyRenderedPosX >= expandcollapseRenderedStartX - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-					&& cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-					if(opts.undoable && !ur)
-						ur = cy.undoRedo({
-							defaultActions: false
-						});
-					if(api.isCollapsible(node))
-						if (opts.undoable){
-							ur.do("collapse", {
-								nodes: node,
-								options: opts
-							});
-						}
-						else
-							api.collapse(node, opts);
-				else if(api.isExpandable(node))
-					if (opts.undoable)
-						ur.do("expand", {
-							nodes: node,
-							options: opts
-						});
-					else
-						api.expand(node, opts);
-					}
-			}
-		});
+        cy.on('mouseover', 'node', data.eMouseOver = function (e) {
+          let node = this;
+          if (isAGroup(node)) {
+            // clear draws if any
+            if ( hoveredGroup && hoveredGroup.id() != node.id() ) {
+
+              refreshCanvasImages();
+            }
+            if (!selectedGroupsContains(node)) {
+              drawExpandCollapseCue(node);
+              hoveredGroup = node;
+            }
```

Comment:
Can the two blocks of code above be simplified? I would think you just need to do the `selectedGroupContains()` check to optionally set the `hoveredGroup`, then always call `refreshCanvasImages()`.